test/new-e2e/tests/agent-platform/common/agent_install.go

```diff
@@ -45,13 +45,13 @@ func CheckSigningKeys(t *testing.T, client *TestClient) {
 // CheckInstallationMajorAgentVersion run tests to check the installation of an agent has the correct major version
 func CheckInstallationMajorAgentVersion(t *testing.T, client *TestClient, expectedVersion string) bool {
 	return t.Run("Check datadog-agent status version", func(tt *testing.T) {
-		versionRegexPattern := regexp.MustCompile(`(?m:^Agent \(v([0-9]).*\)$)`)
+		versionRegexPattern := regexp.MustCompile(`(?m:^(IoT )?Agent \(v([0-9]).*\)$)`)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
We could extract and unit test this regex